core/deployment/pom.xml

```diff
@@ -63,7 +63,11 @@
             <artifactId>quarkus-junit5-internal</artifactId>
             <scope>test</scope>
         </dependency>
-
+        <dependency>
```

Comment:
What's the reason for this?

Reply:
If you try to write an input guard rail test with no memory id annotation we will get: 

```
java.lang.NoClassDefFoundError: io/quarkus/websockets/next/WebSocketConnection

	at io.quarkiverse.langchain4j.websockets.next.runtime.WebSocketConnectionDefaultMemoryIdProvider.getMemoryId(WebSocketConnectionDefaultMemoryIdProvider.java:22)
	at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.memoryId(AiServiceMethodImplementationSupport.java:641)
	at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.doImplement(AiServiceMethodImplementationSupport.java:139)
	at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.implement(AiServiceMethodImplementationSupport.java:122)
	at io.quarkiverse.langchain4j.test.guardrails.InputGuardrailPromptTemplateTest$MyAiService$$QuarkusImpl.sayHiToMyFriendNoMemory(Unknown Source)

````

That was the reason why I included this dependency. You can check other tests; we always have the memory id annotation.

Reply:
You will have to pass the memory id directly.
Look how I implemented the guardrail tests in core.

Reply:
(directly => explicitly)